### PR TITLE
updated model card for Mistral

### PR DESCRIPTION
**Pull request by NahieliV**
This PR updates the model card for Mistral and follows the template outlined in the issue. Please let me know if any changes need to be made. 

Fixes https://github.com/huggingface/transformers/issues/36979

### Before submitting

- [x] This PR improves the docs.

### Who can review?
@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/mistral.md [hunk 1]
```diff
@@ -25,45 +25,45 @@ rendered properly in your Markdown viewer.
 <img alt="SDPA" src="https://img.shields.io/badge/SDPA-DE3412?style=flat&logo=pytorch&logoColor=white">
 </div>
 
-## Overview
+## Model 
 
-Mistral was introduced in the [this blogpost](https://mistral.ai/news/announcing-mistral-7b/) by Albert Jiang, Alexandre Sablayrolles, Arthur Mensch, Chris Bamford, Devendra Singh Chaplot, Diego de las Casas, Florian Bressand, Gianna Lengyel, Guillaume Lample, Lélio Renard Lavaud, Lucile Saulnier, Marie-Anne Lachaux, Pierre Stock, Teven Le Scao, Thibaut Lavril, Thomas Wang, Timothée Lacroix, William El Sayed.
+Mistral is a decoder-only Transformer model that incorporates several architectural choices to enhance its performance. It utilizes Sliding Window Attention, trained with an 8k context length and a fixed cache size, offering a theoretical attention span of up to 128K tokens. Additionally, Mistral-7B employs Grouped Query Attention (GQA), which helps to speed up inference and reduce the cache size. The model also features a Byte-fallback BPE tokenizer, ensuring that characters are never mapped to out-of-vocabulary tokens, thus improving token handling and overall efficiency.
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:30]:
```suggestion
[Mistral](https://huggingface.co/papers/2310.06825) is a 7B parameter language model, available as a pretrained and instruction-tuned variant, focused on balancing 
the scaling costs of large models with performance and efficient inference. This model uses sliding window attention (SWA) trained with a 8K context length and a fixed cache size to handle longer sequences more effectively. Grouped-query attention (GQA) speeds up inference and reduces memory requirements. Mistral also features a byte-fallback BPE tokenizer to improve token handling and efficiency by ensuring characters are never mapped to out-of-vocabulary tokens.
```

File: docs/source/en/model_doc/mistral.md [hunk 2]
```diff
@@ -25,45 +25,45 @@ rendered properly in your Markdown viewer.
...
 
-The introduction of the blog post says:
+Mistral was introduced in [this blogpost](https://mistral.ai/news/announcing-mistral-7b/)
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:32]:
```suggestion
You can find all the original Mistral checkpoints under the [Mistral AI_](https://huggingface.co/mistralai) organization.
```

File: docs/source/en/model_doc/mistral.md [hunk 3]
```diff
@@ -25,45 +25,45 @@ rendered properly in your Markdown viewer.
...
 
-The introduction of the blog post says:
+Mistral was introduced in [this blogpost](https://mistral.ai/news/announcing-mistral-7b/) 
 
-*Mistral AI team is proud to release Mistral 7B, the most powerful language model for its size to date.*
+> [!TIP]
+> Click on the Mistral models in the right sidebar for more examples of how to apply Mistral to different language tasks.
 
-Mistral-7B is the first large language model (LLM) released by [mistral.ai](https://mistral.ai/).
+The example below demonstrates how to generate text with [`Pipeline`] or the [`AutoModel`], and from the command line.
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:37]:
```suggestion
The example below demonstrates how to chat with [`Pipeline`] or the [`AutoModel`], and from the command line.
```

File: docs/source/en/model_doc/mistral.md [hunk 4]
```diff
@@ -25,45 +25,45 @@ rendered properly in your Markdown viewer.
...
 
-`Mistral-7B` is released under the Apache 2.0 license.
+>>> messages = [
+...     {"role": "user", "content": "What is your favourite condiment?"},
+...     {"role": "assistant", "content": "Well, I'm quite partial to a good squeeze of fresh lemon juice. It adds just the right amount of zesty flavour to whatever I'm cooking up in the kitchen!"},
+...     {"role": "user", "content": "Do you have mayonnaise recipes?"}
+... ]
 
-## Usage tips
+chatbot = pipeline("text-generation", model="mistralai/Mistral-7B-Instruct-v0.3")
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:52]:
```suggestion
chatbot = pipeline("text-generation", model="mistralai/Mistral-7B-Instruct-v0.3", torch_dtype=torch.bfloat16, device=0)
```

File: docs/source/en/model_doc/mistral.md [hunk 5]
```diff
@@ -25,45 +25,45 @@ rendered properly in your Markdown viewer.
...
+<hfoption id="AutoModel">
 
 The base model can be used as follows:
 
 ```python
 >>> from transformers import AutoModelForCausalLM, AutoTokenizer
 
->>> model = AutoModelForCausalLM.from_pretrained("mistralai/Mistral-7B-v0.1", device_map="auto")
->>> tokenizer = AutoTokenizer.from_pretrained("mistralai/Mistral-7B-v0.1")
+>>> model = AutoModelForCausalLM.from_pretrained("mistralai/Mistral-7B-v0.3", device_map="auto")
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:65]: Lets just show an example of the instruction-tuned version

File: docs/source/en/model_doc/mistral.md [hunk 6]
```diff
@@ -78,10 +78,11 @@ The base model can be used as follows:
 The instruction tuned model can be used as follows:
 
 ```python
+import torch
 >>> from transformers import AutoModelForCausalLM, AutoTokenizer
 
->>> model = AutoModelForCausalLM.from_pretrained("mistralai/Mistral-7B-Instruct-v0.2", device_map="auto")
->>> tokenizer = AutoTokenizer.from_pretrained("mistralai/Mistral-7B-Instruct-v0.2")
+>>> model = AutoModelForCausalLM.from_pretrained("mistralai/Mistral-7B-Instruct-v0.3", device_map="auto")
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:84]:
```suggestion
>>> model = AutoModelForCausalLM.from_pretrained("mistralai/Mistral-7B-v0.3", torch_dtype=torch.bfloat16, attn_implementation="sdpa", device_map="auto")
```

File: docs/source/en/model_doc/mistral.md [hunk 7]
```diff
@@ -95,9 +96,65 @@ The instruction tuned model can be used as follows:
...
 "Mayonnaise can be made as follows: (...)"
 ```
-
 As can be seen, the instruction-tuned model requires a [chat template](../chat_templating) to be applied to make sure the inputs are prepared in the right format.
 
+<hfoptions id="usage">
+<hfoption id="transformers-cli">
+
+```python
+echo -e "My favorite condiment is" | transformers-cli run --task text-generation --model mistralai/Mistral-7B-v0.3
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:105]:
```suggestion
# pip install -U flash-attn --no-build-isolation
echo -e "My favorite condiment is" | transformers-cli chat --model mistralai/Mistral-7B-v0.3 --torch_dtype auto --device 0 --attn_implementation flash_attention_2
```

File: docs/source/en/model_doc/mistral.md [hunk 8]
```diff
@@ -95,9 +96,65 @@ The instruction tuned model can be used as follows:
...
+
+```python
+echo -e "My favorite condiment is" | transformers-cli run --task text-generation --model mistralai/Mistral-7B-v0.3 
+```
+
+## Shrinking down Mistral using quantization
+
+As the Mistral model has 7 billion parameters, that would require about 14GB of GPU RAM in half precision (float16), since each parameter is stored in 2 bytes. However, one can shrink down the size of the model using [quantization](../quantization.md). If the model is quantized to 4 bits (or half a byte per parameter),that requires only about 3.5GB of RAM.
+
+Quantizing a model is as simple as passing a `quantization_config` to the model. Below, we'll leverage the BitsAndyBytes quantization (but refer to [this page](../quantization.md) for other quantization methods):
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:112]:
```suggestion
Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.

The example below uses [bitsandbytes](../quantization/bitsandbytes) to only quantize the weights to 4-bits.
```

File: docs/source/en/model_doc/mistral.md [hunk 9]
```diff
@@ -95,9 +96,65 @@ The instruction tuned model can be used as follows:
...
+>>> from transformers import AutoModelForCausalLM, AutoTokenizer, BitsAndBytesConfig
+
+>>> # specify how to quantize the model
+>>> quantization_config = BitsAndBytesConfig(
+...         load_in_4bit=True,
+...         bnb_4bit_quant_type="nf4",
+...         bnb_4bit_compute_dtype="torch.float16",
+... )
+
+>>> model = AutoModelForCausalLM.from_pretrained("mistralai/Mistral-7B-Instruct-v0.3", quantization_config=True, device_map="auto")
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:125]:
```suggestion
>>> model = AutoModelForCausalLM.from_pretrained("mistralai/Mistral-7B-Instruct-v0.3", quantization_config=True, torch_dtype=torch.bfloat16, device_map="auto")
```

File: docs/source/en/model_doc/mistral.md [hunk 10]
```diff
@@ -95,9 +96,65 @@ The instruction tuned model can be used as follows:
...
+
+visualizer = AttentionMaskVisualizer("mistralai/Mistral-7B-Instruct-v0.3")
+visualizer("Do you have mayonnaise recipes?")
+```
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/mistral-attn-mask.png"/>
+</div>
+
+## Notes
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:156]: I think we've covered pretty much everything so no need for a `##Notes` section and can remove the other text under it.

**Review by stevhliu — COMMENTED** [review]
Very nice and clean! 👏

**Comment by NahieliV** [comment]
@stevhliu , I have incorporated your suggestions. Should be good now!

File: docs/source/en/model_doc/mistral.md [hunk 11]
```diff
@@ -25,45 +25,45 @@ rendered properly in your Markdown viewer.
 <img alt="SDPA" src="https://img.shields.io/badge/SDPA-DE3412?style=flat&logo=pytorch&logoColor=white">
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:25]: These badges should go at the top above `# Mistral`

File: docs/source/en/model_doc/mistral.md [hunk 12]
```diff
@@ -25,45 +25,45 @@ rendered properly in your Markdown viewer.
...
+<hfoption id="AutoModel">
 
 The base model can be used as follows:
 
 ```python
 >>> from transformers import AutoModelForCausalLM, AutoTokenizer
 
->>> model = AutoModelForCausalLM.from_pretrained("mistralai/Mistral-7B-v0.1", device_map="auto")
->>> tokenizer = AutoTokenizer.from_pretrained("mistralai/Mistral-7B-v0.1")
+>>> model = AutoModelForCausalLM.from_pretrained("mistralai/Mistral-7B-v0.3", device_map="auto")
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:65]: Remove the code snippet for the base version

File: docs/source/en/model_doc/mistral.md [hunk 13]
```diff
@@ -95,60 +96,19 @@ The instruction tuned model can be used as follows:
 >>> tokenizer.batch_decode(generated_ids)[0]
 "Mayonnaise can be made as follows: (...)"
 ```
-
 As can be seen, the instruction-tuned model requires a [chat template](../chat_templating) to be applied to make sure the inputs are prepared in the right format.
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:99]: Can remove this as well

File: docs/source/en/model_doc/mistral.md [hunk 14]
```diff
@@ -95,60 +96,19 @@ The instruction tuned model can be used as follows:
...
->>> prompt = "My favourite condiment is"
-
->>> model_inputs = tokenizer([prompt], return_tensors="pt").to("cuda")
->>> model.to(device)
-
->>> generated_ids = model.generate(**model_inputs, max_new_tokens=100, do_sample=True)
->>> tokenizer.batch_decode(generated_ids)[0]
-"My favourite condiment is to (...)"
+# pip install -U flash-attn --no-build-isolation
+echo -e "My favorite condiment is" | transformers-cli chat --model mistralai/Mistral-7B-v0.3 --torch_dtype auto --device 0 --attn_implementation flash_attention_2
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:106]:
```suggestion
echo -e "My favorite condiment is" | transformers-cli chat --model_name_or_path mistralai/Mistral-7B-v0.3 --torch_dtype auto --device 0 --attn_implementation flash_attention_2
```

File: docs/source/en/model_doc/mistral.md [hunk 15]
```diff
@@ -25,45 +25,45 @@ rendered properly in your Markdown viewer.
...
+chatbot = pipeline("text-generation", model="mistralai/Mistral-7B-Instruct-v0.3", torch_dtype=torch.bfloat16, device=0)
+chatbot(messages)
 
-The Mistral team has released 3 checkpoints:
+```
 
-- a base model, [Mistral-7B-v0.1](https://huggingface.co/mistralai/Mistral-7B-v0.1), which has been pre-trained to predict the next token on internet-scale data.
-- an instruction tuned model, [Mistral-7B-Instruct-v0.1](https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.1), which is the base model optimized for chat purposes using supervised fine-tuning (SFT) and direct preference optimization (DPO).
-- an improved instruction tuned model, [Mistral-7B-Instruct-v0.2](https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.2), which improves upon v1.
+<hfoptions id="usage">
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:57]: The `<hfoptions>` should be used like:

```md
<hfoptions id="usage">
<hfoption id="Pipeline">

...

</hfoption>
<hfoption id="AutoModel">

...

</hfoption>
<hfoption id="transformers-cli">

</hfoption>
</hfoptions>
```

File: docs/source/en/model_doc/mistral.md [hunk 16]
```diff
@@ -78,10 +78,11 @@ The base model can be used as follows:
 The instruction tuned model can be used as follows:
 
 ```python
+>>> import torch
 >>> from transformers import AutoModelForCausalLM, AutoTokenizer
 
->>> model = AutoModelForCausalLM.from_pretrained("mistralai/Mistral-7B-Instruct-v0.2", device_map="auto")
->>> tokenizer = AutoTokenizer.from_pretrained("mistralai/Mistral-7B-Instruct-v0.2")
+>>> model = AutoModelForCausalLM.from_pretrained("mistralai/Mistral-7B-v0.3", torch_dtype=torch.bfloat16, attn_implementation="sdpa", device_map="auto")
```
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:84]:
```suggestion
>>> model = AutoModelForCausalLM.from_pretrained("mistralai/Mistral-7B-Instruct-v0.3", torch_dtype=torch.bfloat16, attn_implementation="sdpa", device_map="auto")
```

File: docs/source/en/model_doc/mistral.md (hunk 14 above)
Comment (stevhliu) [inline docs/source/en/model_doc/mistral.md:106]:
```suggestion
echo -e "My favorite condiment is" | transformers-cli chat --model mistralai/Mistral-7B-Instruct-v0.3 --torch_dtype auto --device 0 --attn_implementation flash_attention_2
```

**Review by stevhliu — COMMENTED** [review]
Thanks, just several more formatting changes!

File: docs/source/en/model_doc/mistral.md (hunk 12 above)
Comment (NahieliV) [reply docs/source/en/model_doc/mistral.md:65]: Sorry, I don't understand what you mean :(. Do you mean replacing the base model by the instruction tuned one?  We need the tokenizer so I can't accept the proposed change.

Comment (NahieliV) [reply docs/source/en/model_doc/mistral.md:65]: Never mind, I will remove the whole code snippet for the base model.

Comment (stevhliu) [reply docs/source/en/model_doc/mistral.md:65]: Yeah sorry I wasn't clear enough! We can remove the whole code snippet for the base model and focus on the instruction-tuned one. We don't need to show an example with both :)

**Review by stevhliu — APPROVED** [review]
Thanks for your contribution! 🚀
